### ISSUE TITLE
Lima prefers and fails with IPv6 to access registry after upgrade to 0.7.1 

### ISSUE THREAD

**Issue by romanwoessner** [rancher-sandbox/rancher-desktop#1193]
### Rancher Desktop Version

0.7.1

### Rancher Desktop K8s Version

1.22.5

### What operating system are you using?

macOS

### Operating System / Build Version

macOS Monterey 12.1

### What CPU architecture are you using?

x64

### Actual Behavior

I am using RD with dockerd(moby).
After updating from version 0.7.0 to 0.7.1, I can no longer pull images from our company registry:
```
❯ docker pull registry.mycompany.net/awsome-image:1.0.0
Error response from daemon: Get "https://registry.mycompany.net/v2/": net/http: TLS handshake timeout
```
To see what is actually going on, I have curled the registry API from inside the lima VM:
```
❯ curl -v https://registry.mycompany.net/v2/
*  Trying 2001:67c:2388:104::a08:c35:443...
* Connected to registry.mycompany.net (2001:67c:2388:104::a08:c35) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
* CAfile: /etc/ssl/certs/ca-certificates.crt
* CApath: none
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
…
(nothing else happens)
```
As you can see the hostname resolves to an IPv6 address. A request with curl -4 https://... (IPv4) works fine.
The hostname of our registry actually resolves to both IPv4 and IPv6:
```
❯ nslookup registry.mycompany.net
Server:		192.168.5.3
Address:	192.168.5.3:53

Non-authoritative answer:
Name:	registry.mycompany.net
Address: 2001:67c:2388:104::a08:c35

Non-authoritative answer:
Name:	registry.mycompany.net
Address: 10.8.12.53
```

As a test and to exclude changes in our infrastructure, I have carried out a rollback to 0.7.0.
`docker pull`  and `curl` do not fail there. They use the IPv4 address to access the registry:
```
❯ curl -v https://registry.mycompany.net
*   Trying 2001:67c:2388:104::a08:c35:443...
*   Trying 10.8.12.53:443...
* Connected to registry.mycompany.net (10.8.12.53) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: none
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS handshake, Finished (20):
```
I guess it is somehow related to the changed mechanism on how RD tries to find the correct network bridge now: https://github.com/rancher-sandbox/rancher-desktop/compare/v0.7.0...v0.7.1

### Steps to Reproduce

I guess you need a registry with a hostname which resolves to IPv6 and IPv4.
Update to 0.7.1 and try to login or pull images from your registry.

### Result

```
Error response from daemon: Get "https://registry.mycompany.net/v2/": net/http: TLS handshake timeout
```

### Expected Behavior

Lima should stick to IPv4 to access registries.

**Comment by jandubois** [comment]
It is not clear to my why 0.7.0 would just use IPv4, but 0.7.1 would use IPv6.

But to fix the problem I've changed the host resolver in lima to not return any IPv6 addresses by default. This change should be in the 1.0.0-beta release due early next week. Please test and confirm that the issue is gone (or re-open the issue, if it is not).

**Comment by romanwoessner** [comment]
I can confirm that the problem is fixed for me with version 1.0.0-beta.1.
Thanks a lot!
